### ISSUE TITLE
Toggling focusEnable behaves differently between lineChart and linePlusBarChart

### ISSUE THREAD

**Issue by MikeMatusz** [novus/nvd3#1859]
In the following example using linePlusBarChart, toggling focusEnable does not clear the selected range and works the way I would expect it to:
http://nvd3-community.github.io/nvd3/examples/linePlusBarChart.html

If you modify that example to use lineChart:
https://plnkr.co/edit/3PMkJeOvCsYoO81L3Xyz?p=preview

Now, toggling focusEnable does clear the selected range, and also fails to actually hide the focus chart.  That latter part is definitely an issue, but the former is less clear.  Some people might actually be using the chart expecting that behavior, but in my case, I'd like it to behave the same as linePlusBarChart.  I would be willing to submit a PR for this, but should I add an additional option for lineChart to opt-in to not clearing the brush extent when the focus chart is hidden so the default behavior remains unchanged?

**Comment by liquidpele** [comment]
It's not supposed to clear the focus range.  Also, your plunker doesn't remove the focus if you toggle it again.

**Comment by MikeMatusz** [comment]
Maybe I didn't explain myself properly, but based on your response, I think this _is_ a valid issue.

It's not my plunker that isn't removing the focus chart, it's a problem with the library.  I've done quite a bit of debugging before submitting the issue here.  That's one half of this issue.

Also, it _is_ clearing the focus range, so it sounds like you're agreeing with me, and I'm not sure why this issue was closed.

Try this:
Toggle the focus chart ON
Select a subset of the full range on the focus chart.
Toggle the focus chart OFF

Expected Behavior:
The domain of the main chart should remain filtered and not reset
Focus chart is hidden

The linePlusBarChart example exhibits these behaviors, which you just agreed are the correct ones.
The lineChart fails on both.  The domain of the main chart is resetting when focusEnable is set to false.

Please reopen this issue, or provide either a working example or a better explanation of why it was closed.

**Comment by liquidpele** [comment]
Well, I'm in the office now and have had my coffee, and I'm not sure what I was thinking this morning :p

**Comment by MikeMatusz** [comment]
Haha, thanks!  So I have modified the code locally (I'll create a fork soon), but it's basically removing a few blocks of code from lineChart and adding the display none/initial that linePlusBarChart uses.  My concern is that someone might currently be using it and expecting the domain to reset (i.e. this would be a breaking change) so I was wondering if I should create a setting to opt-in to the behavior.

If you'd prefer to see what's being removed before making that determination, let me know and I'll submit an initial PR without the setting and let you decide.

**Comment by liquidpele** [comment]
No, it really shouldn't reset.  Can't really worry about it if people are relying on a bug...
